internal/config/filter_test.go

```diff
@@ -50,47 +50,113 @@ func TestEgressFilter_Bind(t *testing.T) {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           Client: GRPCClient{},
-		           DistanceFilters: nil,
-		           ObjectFilters: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           Client: GRPCClient{},
-		           DistanceFilters: nil,
-		           ObjectFilters: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			return test{
+				name: "return EgressFilter when the bind successes",
+				fields: fields{
+					DistanceFilters: []string{
+						"192.168.1.2",
+					},
+					ObjectFilters: []string{
+						"192.168.1.3",
+					},
+				},
+				want: want{
+					want: &EgressFilter{
+						DistanceFilters: []string{
+							"192.168.1.2",
+						},
+						ObjectFilters: []string{
+							"192.168.1.3",
+						},
+					},
+				},
+			}
+		}(),
+		func() test {
+			return test{
+				name: "return EgressFilter when the bind successes and the Client is not nil",
+				fields: fields{
+					DistanceFilters: []string{
+						"192.168.1.2",
+					},
+					ObjectFilters: []string{
+						"192.168.1.3",
+					},
+					Client: new(GRPCClient),
+				},
+				want: want{
+					want: &EgressFilter{
+						DistanceFilters: []string{
+							"192.168.1.2",
+						},
+						ObjectFilters: []string{
+							"192.168.1.3",
+						},
+						Client: &GRPCClient{
+							ConnectionPool: new(ConnectionPool),
+							DialOption: &DialOption{
+								Insecure: true,
+							},
+							TLS: new(TLS),
+						},
+					},
+				},
+			}
+		}(),
+		func() test {
+			m := map[string]string{
+				"DISTANCE_FILTERS": "192.168.1.2",
+				"OBJECT_FILTERS":   "192.168.1.3",
+			}
+			return test{
+				name: "return EgressFilter when the bind successes and the data is loaded from the environment variable",
+				fields: fields{
+					DistanceFilters: []string{
+						"_DISTANCE_FILTERS_",
+					},
+					ObjectFilters: []string{
+						"_OBJECT_FILTERS_",
+					},
+				},
+				beforeFunc: func(t *testing.T) {
+					t.Helper()
+					for k, v := range m {
+						if err := os.Setenv(k, v); err != nil {
+							t.Fatal(err)
+						}
+					}
+				},
+				afterFunc: func(t *testing.T) {
+					t.Helper()
+					for k := range m {
+						if err := os.Unsetenv(k); err != nil {
+							t.Fatal(err)
+						}
+					}
+				},
+				want: want{
+					want: &EgressFilter{
+						DistanceFilters: []string{
+							"192.168.1.2",
+						},
+						ObjectFilters: []string{
+							"192.168.1.3",
+						},
+					},
+				},
+			}
+		}(),
 	}
 
-	for _, tc := range tests {
-		test := tc
+	for _, test := range tests {
 		t.Run(test.name, func(tt *testing.T) {
-			tt.Parallel()
```

Comment:
should we use parallel here?
if you wanted to avoid the conflict on the environment variable, I think we can use different env variable name for different test case.

Reply:
thank you. It looks good to me.
I will fix

Reply:
@kevindiu 
https://github.com/vdaas/vald/pull/1125/commits/004087ee4f6c33de94897fc580951b1624c62538